app/uk/gov/hmrc/economiccrimelevyaccount/controllers/AccountController.scala

```diff
@@ -48,20 +49,31 @@ class AccountController @Inject() (
     enrolmentStoreProxyService.getEclRegistrationDate(request.eclRegistrationReference).flatMap { registrationDate =>
       obligationDataConnector
         .getObligationData()
-        .map { o =>
+        .flatMap { o =>
           auditAccountViewed(o)
-
           o match {
             case Some(obligationData) =>
-              Ok(
-                view(
-                  request.eclRegistrationReference,
-                  ViewUtils.formatLocalDate(registrationDate),
-                  getLatestObligation(obligationData)
+              financialDataService.latestFinancialObligation.map { financialData =>
```

Comment:
Consider this to cut down duplication

`def onPageLoad: Action[AnyContent] = authorise.async { implicit request =>
    enrolmentStoreProxyService.getEclRegistrationDate(request.eclRegistrationReference).flatMap { registrationDate =>
      obligationDataConnector
        .getObligationData()
        .flatMap { o =>
          auditAccountViewed(o)
          financialDataService.latestFinancialObligation.map { financialData =>
            Ok(
              view(
                request.eclRegistrationReference,
                ViewUtils.formatLocalDate(registrationDate),
                o match {
                  case Some(obligationData) => getLatestObligation(obligationData)
                  case None => None
                },
                financialData
              )
            )
          }
        }
    }
  }`

Reply:
Nice catch, thanks :)